src/GGLSetup.py

```diff
@@ -208,13 +273,23 @@ def _install_build_dependencies() -> bool:
             "liburiparser-dev", "cgroup-tools"
         ]
 
-        # Update package list
-        update_command = ["sudo", "apt", "update"]
-        subprocess.run(update_command, check=True)
+        # Check which packages are missing
+        check_cmd = ["dpkg", "-l"] + packages
+        result = subprocess.run(check_cmd, capture_output=True, text=True)
+
+        missing_packages = []
+        for pkg in packages:
+            if pkg not in result.stdout or f"ii  {pkg}" not in result.stdout:
+                missing_packages.append(pkg)
+
+        if not missing_packages:
+            print("All dependencies already installed")
+            return True
 
-        # Install packages
-        install_command = ["sudo", "apt", "install", "-y"] + packages
-        subprocess.run(install_command, check=True)
+        # Update and install only missing packages
+        subprocess.run(["sudo", "apt", "update"], check=True)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method is running with elevated privileges. Privilege elevation occurs when you access resources with an extended set of permissions. When unattended, these privileges might allow attackers to compromise the system. Do not use elevated privileges when they can be avoided.

[Learn more](https://cwe.mitre.org/data/definitions/269.html)

Similar issue at line number 291.